spec/features/user_google_login_spec.rb

```diff
@@ -48,6 +48,39 @@
               end
             end
           end
+
+          context 'when no User or AdminUser is logged in' do
+            before { Devise.sign_out_all_scopes }
+
+            context 'when there is an AdminUser and a User with the same stubbed_user_email' do
+              before { AdminUser.find_or_create_by!(email: user.email) }
+
+              context 'when the AdminUser has logged in' do
+                before do
+                  visit(new_admin_user_session_path)
+
+                  expect(page).to have_current_path(new_admin_user_session_path)
+
+                  click_on('Sign in with Google')
+
+                  expect(page).to have_text('You are beautiful')
+                end
+
+                context 'when the user tries to go to the My Account page' do
+                  it 'redirects the user to log in and, after successful login, redirects to the My Account page' do
+                    visit(my_account_path)
+
+                    expect(page).to have_current_path(new_user_session_path)
+
+                    click_on('Sign in with Google')
+
+                    expect(page).to have_current_path(my_account_path)
+                    expect(page).to have_text('My Account')
```

Comment:
Prior to this change, this spec would fail by redirecting the User to `/admin` after logging in as a User to access a User-requiring route. Now, we redirect to the intended User page (My Account, in this example), as intended.